frame-metadata/src/lib.rs

```diff
@@ -20,14 +20,32 @@
 #![cfg_attr(not(feature = "std"), no_std)]
 #![warn(missing_docs)]
 
+#[cfg(all(
+	feature = "full_derive",
+	any(
+		feature = "v13",
+		feature = "v12",
+		feature = "v11",
+		feature = "v10",
+		feature = "v9",
+		feature = "v8",
+		feature = "legacy"
+	)
+))]
+compile_error!("feature \"full_derive\" cannot be enabled for metadata prior to version v14");
+
```

Comment:
I have double-checked, I thought that the old metadata versions are not being completely supported in terms of decoding and serializing. However, it seems that they are. If we really want to do an upstream PR, I believe we can't neglect the old ones. So I need to ask you to repeat the pattern from metadata v14 for the old versions.

Reply:
Metadata prior V14 do not support `Decode` if no_std.  See https://github.com/integritee-network/frame-metadata/blob/d241ab9c1ba89e119263812f963a601be158e661/frame-metadata/src/decode_different.rs#L29 
For now the std flag set "full" derives in metadata prior V14. Do we really want to differentiate between the derive levels or between the technical possible ones?  
We added the full-derive flag for v14 to get the derives decode, serialize,... in no_std as they are technically feasible with scale-info. For the metadata prior V14 we could have the Serialize in no_std, but Decode only in std. Does it make sense to have this differentiation?

Reply:
Ahbok, if it is like that, I agree with you. Let's leave it like that. Sorry for that!

Reply:
The changes in this PR are wrong. So if we only change V14, please check https://github.com/integritee-network/frame-metadata/pull/2